Net/SMTP.php

```diff
@@ -1007,8 +1007,29 @@ protected function authGSSAPI($uid, $pwd, $authz = '')
     public function authXOAuth2($uid, $token, $authz, $conn)
     {
         $auth = base64_encode("user=$uid\1auth=$token\1\1");
-        if (PEAR::isError($error = $this->put('AUTH', 'XOAUTH2 ' . $auth))) {
-            return $error;
+
+        // Maximum length of the base64-encoded token to be sent in the initial response is 497 bytes, according to
+        // RFC 4954 (https://datatracker.ietf.org/doc/html/rfc4954); for longer tokens an empty initial
+        // response MUST be sent and the token must be sent separately
+        // (497 bytes = 512 bytes /SMTP command length limit/ - 13 bytes /"AUTH XOAUTH2 "/ - 2 bytes /CRLF/)
+        if (strlen($auth) <= 497) {
+            if (PEAR::isError($error = $this->put('AUTH', 'XOAUTH2 ' . $auth))) {
+                return $error;
+            }
+        } else {
+            if (PEAR::isError($error = $this->put('AUTH', 'XOAUTH2'))) {
+                return $error;
+            }
+
+            // server is expected to respond with 334
+            if (PEAR::isError($error = $this->parseResponse(334))) {
```

Comment:
This should return 334 and empty message. I wonder if we should somehow test that the message from server is empty?